File: admin-backend/MyBlogAdminService/Models/Post.cs
using System.Collections;
using System.ComponentModel.DataAnnotations;

namespace MyBlogAdminService.Models
{
    public class Post
    {
        [Key]
        public int Id { get; set; }
        public string? ImagePath { get; set; }

        [Required(ErrorMessage = "Title is required.")]
        public string? Title { get; set; }

        [Required(ErrorMessage = "Content is required.")]
        public string? Content { get; set; }

        public DateTime? Created { get; set; }

        public ICollection<Category>? Categories { get; set; } = new List<Category>();

        public ICollection<Tag>? Tags { get; set; } = new List<Tag>();
    }
}

File: admin-frontend/src/models/Post.js
export class PostCreateDto {
  constructor(title = '', content = '', categoryIds = [], tagIds = []) {
    this.title = title
    this.content = content
    this.categoryIds = categoryIds
    this.tagIds = tagIds
  }
}

export class PostUpdateDto {
  constructor(id = 0, title = '', content = '', categoryIds = [], tagIds = []) {
    this.id = id
    this.title = title
    this.content = content
    this.categoryIds = categoryIds
    this.tagIds = tagIds
  }
}

export class PostResponseDto {
  constructor(id = 0, imagePath = '' ,title = '', content = '', categories = [], tags = []) {
    this.id = id
    this.imagePath = imagePath
    this.title = title
    this.content = content
    this.categories = categories
    this.tags = tags
  }
}
